app/pages/lab/actions/workflow.js

```diff
@@ -3,12 +3,23 @@ var apiClient = require('panoptes-client/lib/api-client');
 var workflows = apiClient.type('workflows');
 
 var workflowActions = {
-  createWorkflowForProject: function(projectID, workflowData) {
+  createWorkflowForProject: function(project, workflowData) {
     var allWorkflowData = Object.assign({
-      links: {project: projectID}
+      links: {project: project.id}
     }, workflowData);
 
     return workflows.create(allWorkflowData).save();
+  },
+  copyWorkflowForProject: async function(project, workflowData) {
+    await project.addLink('workflows', [workflowData.id]);
+    project.uncacheLink('workflows');
+    // TODO: does this sort work for this query?
+    const newWorkflows = await project.get('workflows', { sort: '-updated_at' });
```

Comment:
Currently this doesn't work, we can look to add it if it's useful, https://github.com/zooniverse/panoptes/blob/5b0cbf938e1e33830ec4204778aa0c6c9f5030bc/app/serializers/workflow_serializer.rb#L21

Alternatively we can modify the project response object to include some data on the copied project if that's easier,
i.e. something like
```
"projects": [{
"id": 1,
....
"copied_workflow_id": $id-of-newly-copied-workflow
}
# or in the links object of the resource response etc.
```
What do you think?

Reply:
That sounds useful. According to the API docs, the new link should be in the links object returned from the POST request, but I couldn’t see it in `links.workflows`. 

https://panoptes.docs.apiary.io/#reference/projects/project-create-links/add-a-link

When I refreshed the page, to force a new load of the project from the API, the link was there but at a random position in the array. If possible, it would be useful if the serialiser sent the links back in creation order, or ordered by workflow ID (which might be the same thing.) In that case, I could reliably pluck the last ID from the array each time. 

Subject sets have the same problem so I’ll have a look at subject set creation to see how that works. I remember `addLink` being tricky when I worked on IIIF manifests.

Reply:
This query is also expensive on projects with large workflows. My test project has 25 workflows. I think Davy Notebooks was planning on having up to 80 workflows (though they might have switched to using a single workflow with many subject sets.)

Reply:
Now that I write that, in many cases a project will want to adds new subject set rather than cloning a workflow. Worth checking with Sam as to exactly what this button is used for.

Reply:
I can intervene on specific resources in the serializer to say sort the workflow links by the id (i.e. in desc order) so you can quickly identify the 'latest' one. However with large payloads that does add a cost (most likely small) on each serialiser query.

For the copy workflow use case, i like your idea of doing a set diff of the links array before and after. That should uniquely identify the copied workflow. Failing that I can even add a new linked resource on this one end point like i added above, `{ copied_workflow: '$id' }` if that is easier. 

Let me know what you find regarding the `subject sets` creation behaviours and if we can generalize a pattern or just deal with this one use case.

Reply:
I think this will fix the issue of the update links response not including the newly linked resources, https://github.com/zooniverse/panoptes/pull/3968

once that lands the relevant resource links response can be diffed with the previous links before the update to determine the newly added resource.

Reply:
Cool. Right now, when I difference the new and old project links, they're both the same. This is after making a fresh project request too. Weirdly, If I go to the project builder workflows page and refresh the page, the new workflow is loaded then.

<img width="1138" alt="Screenshot of the workflow copy button showing a 'no new workflow links on project.' error." src="https://user-images.githubusercontent.com/59547/195556249-0881d0cf-5449-4565-a834-ca3850adeb61.png">

Reply:
I'll push that change up, so that it's available on staging.

Reply:
I'll get that fix out to staging as soon as it's been reviewed